### PR TITLE
[GEOT-7081] Choose correct EPSG library

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-7081](https://osgeo-org.atlassian.net/browse/GEOT-7081) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Fixes a problem with choosing correct EPSG-library when urn specifies a version.
The main solution is to give VERSION as a hint to each EPSG-library involved, in this case gt-epsg-wkt.

I've added some logging within AuthorityFactoryAdaptor which I thought were useful.

There were code within DefaultAuthorityFactory that I think made the whole process less intuitive. I think what that code tried to solve, are solved by the FallbackAuthorityFactory.

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [X] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [X] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Review by mprins — CHANGES_REQUESTED** [review]
this is missing documentation update; docs state there must be only one EPSG jar

**Comment by roarbra** [comment]
Removed the comment about throwing FactoryException, which is wrong. Replaced it with a comment about it causing conflict, which has been used earlier.

File: modules/library/referencing/src/main/java/org/geotools/referencing/DefaultAuthorityFactory.java
Comment (aaime) [inline modules/library/referencing/src/main/java/org/geotools/referencing/DefaultAuthorityFactory.java:94]: So... this change is giving me nightmares. Not because I think it's wrong, per se, but because the referencing subsystem is known for reconfiguring itself dynamically, scanning for factories makes new factories pop up or re-order themselves (don't ask, I don't remember why this happens). That's why there was this loop, which would otherwise seem nonsensical.

That said, all tests are passing. On the other end, the whole ticket is addressing an IDE setup deficiency, not an actual runtime issue.
I'm going to wait until 27-RC is cut, and then merge when main becomes 28.x. This should give us six months to see f there are issues with the change that tests are not finding, and revert before this gets in the user's face.

**Comment by roarbra** [comment]
Thanks!

**Pull request merged**
